app/models/manageiq/providers/amazon/storage_manager/s3/cloud_object_store_container.rb

```diff
@@ -24,7 +24,11 @@ class ManageIQ::Providers::Amazon::StorageManager::S3::CloudObjectStoreContainer
 
   def provider_object(connection = nil)
     connection ||= ext_management_system.connect
-    connection.bucket(ems_ref)
```

Comment:
What if `connection` parameter is given? Should it fail in case it is not correct for the corresponding bucket?